docs/card-rendering.md

```diff
@@ -0,0 +1,20 @@
+# Card Rendering
+
+To render a card, the primary step is to obtain a Glimmer component identified by the template tags within the card. A card may contain several Glimmer components, including fields and other miscellaneous components, that are categorized into three formats: isolated, embedded, and edit. These formats determine which component should be rendered. If there are no components available for the isolated or edit formats, the [default format](../packages/base/default-card-component.gts) is used instead. The Glimmer components themselves are building blocks that can be used to create many other things.
+
+## Format
+- Isolated format renders a card as a root component.
+- Embedded format is useful when a card is embedded within another card.
+- Edit format is used when the user wants to update the value of the card.
+
+## Field
+A card object can serve as a field within another card object. To establish a card as a field for another card, you can use the @field decorator and specify one of the following field types: `contains`, `containsMany`, `linksTo`, or `linksToMany`. The main difference between a contained field and a linked field is that a linked field refers to another card instance in the realm, whereas a contained field creates a new card instance within the root component of the original card. In other words, a linked field has an identity to refer to, while a contained field doesn't. In edit mode, a contained field will provide inputs for updating its value, while a linked field will provide a modal to search for an existing card instance.
+
+## Rendering process
+1. Wrapping a card in Box class as a root component.
+2. Getting a glimmer component based on format.
+3. Getting fields from the Box class, including field's component.
+4. Using fields and box class to instantiate the glimmer component.
+
+## Re-rendering process
+Our approach leverages the glimmer invalidation system to trigger card re-rendering. To achieve this, we rely on [TrackedWeakMap implementation](../packages/base/card-api.gts#L136-L138), where each card instance is used as a key. Whenever we call the .get() method on an item in this map, it automatically prompts glimmer to update the card.
```

Comment:
I think it would be more correct to say that we call .get() on this map to entangle a field into Glimmer's update tracking, and call .set() in order to signal to Glimmer that a field needs to be re-rendered.